pkg/workflow/workflow.go

```diff
@@ -458,8 +487,31 @@ func (e *engine) getMaxBackoffWaitTime() int {
 	return MaxWorkflowWaitBackoffTime
 }
 
+func (e *engine) getNextTimeout() int64 {
+	max := time.Duration(1<<63 - 1)
+	min := time.Duration(1<<63 - 1)
+	now := time.Now()
+	for _, step := range e.status.Steps {
+		if step.Phase == common.WorkflowStepPhaseRunning {
+			if timeout, ok := e.stepTimeout[step.Name]; ok {
+				duration := timeout.Sub(now)
+				if duration < min {
+					min = duration
+				}
+			}
+		}
+	}
+	if min == max {
+		return -1
+	}
+	if min.Seconds() < 1 {
+		return minWorkflowBackoffWaitTime
+	}
+	return int64(math.Ceil(min.Seconds()))
```

Comment:
why need math.Ceil here?

Reply:
if the `min.Seconds == 1.2`, the next requeue time should be 2 seconds later.